crates/bevy_asset/src/server/mod.rs

```diff
@@ -1339,6 +1339,17 @@ pub enum LoadState {
     Failed(Box<AssetLoadError>),
 }
 
+impl LoadState {
+    /// Returns `true` if this instance is [`LoadState::Loading`]
+    pub fn is_loading(&self) -> bool {
+        matches!(self, Self::Loading)
+    }
+    /// Returns `true` if this instance is [`LoadState::Loaded`]
+    pub fn is_loaded(&self) -> bool {
+        matches!(self, Self::Loaded)
+    }
+}
```

Comment:
Nit: while we're at it, should we add methods for the other variants?

Reply:
I didn't want to add a variant for `NotLoaded`, since people might incorrectly assume it's complementary to `is_loaded`. I'm neutral, leaning no, on adding `is_failed`, since I suspect in most cases people will actually want to be checking for `!is_loaded && !is_loading`

Reply:
I think `is_failed` could be useful. But no need to block on adding it or not haha it can always be saved until someone needs it